### PR TITLE
Video wall

### PR DESCRIPTION
**Pull request by helge000**
## Add VIDEO_WALL related commands

I only need the trivial `VIDEO_WALL_ENABLE`, maybe you can have a look over the second commit: VIDEO_WALL_MODEL.
This is more complicated; maybe you can have a look how you like this implemented?

### CODE REVIEW

File: samsung_mdc/fields.py [hunk 1]
```diff
@@ -151,3 +151,16 @@ def pack(self, data):
         if not len(rv) == 4 or not all(0 <= x < 256 for x in rv):
             raise ValueError('Invalid IP address', data)
         return rv
+
+
+class VideoWall(Field):
```
Comment (helge000) [inline samsung_mdc/fields.py:156]: I think I need a new field for this
Video Wall Model def
:

File: samsung_mdc/utils.py [hunk 2]
```diff
@@ -52,3 +52,11 @@ def repr_hex(value):
 
 def parse_hex(value):
     return value and bytes(int(x, 16) for x in value.split(':')) or b''
+
+
+def parse_videowall(value):
```
Comment (helge000) [inline samsung_mdc/utils.py:57]: And add parse and pack methods here; or do you want them in the field?

File: samsung_mdc/utils.py [hunk 3]
```diff
@@ -52,3 +52,20 @@ def repr_hex(value):
...
+    return divmod(value, 1<<4)[::-1]
+
+
+def pack_videowall(value):
+    """
+    Takes tuple of length 2 and combines its numbers as 2x4 to one 8 bit value
+    Reverses representation to more intuitive X, Y
+    """
+    coords = repr_hex(value)
+    return coords[4] + coords[1]
```
Comment (helge000) [inline samsung_mdc/utils.py:71]: This is not very neat ... but get's the job done. Do you have a better idea?

**Comment by helge000** [comment]
I tested everything from the first commit; the second one should also work now.

File: samsung_mdc/utils.py [hunk 4]
```diff
@@ -52,3 +52,20 @@ def repr_hex(value):
...
 def parse_hex(value):
     return value and bytes(int(x, 16) for x in value.split(':')) or b''
+
+
+def parse_videowall(value):
+    """
+    Splits 8 byte 'coordinates' to 2x4 byte x/y tuple
+    The expected representation is Y, X
+    """
+    return divmod(value, 1<<4)[::-1]
```
Comment (helge000) [inline samsung_mdc/utils.py:62]: This will only work if the `value` I get here is hex in the form `0xff`. Is this the case?

**Comment by vgavro** [comment]
@helge000 Thanks for PR, great work!
I merged it and released in 1.8.0 version with some fixes:
1. There is no need to keep SERIAL of video wall in same field as coordinates. So I moved it to separate field.
2. pack_videowall implementation wasn't clear for me - so I rewrite it - hope it still works the same)
3. What was the reason for limit for x * y < 90 and serial < 90? I believe this should be 100 or 99 for 10x10 video wall? There is no such limit in latest protocol documentation (see MDC-Protocol.pdf from root repository), there is 225 limit (15x15 max), but I believe it's ok to leave such errors to samsung anyway (to NAK error), or add separate error for exceeding limit (don't mix it with wrong tuple length i mean) - but really I wouldn't bother).

Please test it on your video wall and confirm everything is ok, or feel free to create new PR if you think I missed something!

**Pull request merged**